### PR TITLE
Add Archive without data

### PR DESCRIPTION
**Pull request by be-marc**
```r
library(paradox)

fun = function(xs) {
  c(y = - (xs[[1]] - 2)^2 - (xs[[2]] + 3)^2 + 10)
}

domain = ParamSet$new(list(
  ParamDbl$new("x1", -10, 10), 
  ParamDbl$new("x2", -5, 5)
))

codomain = ParamSet$new(list(
  ParamDbl$new("y", tags = "maximize")
))

obfun = ObjectiveRFun$new(
  fun = fun,
  domain = domain,
  codomain = codomain, 
  properties = "deterministic", 
  check_values = FALSE
)

terminator = trm("evals", n_evals = 10000)

instance = OptimInstanceSingleCrit$new(
  objective = obfun, 
  terminator = terminator,
  keep_evals = "best",
  check_values = FALSE
)

optimizer = opt("random_search", batch_size = 10)

optimizer$optimize(instance)
```

### CODE REVIEW

File: R/ArchiveBest.R [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+#' parameters are ignored and some methods do nothing. The archive still works
+#' with [TerminatorClockTime], [TerminatorEvals], [TerminatorNone] and
+#' [TerminatorEvals].
+#'
+#' @template param_codomain
+#' @template param_search_space
+#' @template param_xdt
+#' @template param_ydt
+#' @export
+ArchiveBest = R6Class("ArchiveBest",
```
Comment (jakob-r) [inline R/ArchiveBest.R:15]: `ArchivePartial`? Hmm. but it should be consistent with `keep_evals == "best"`. Maybe the name is ok after all.

File: R/ArchiveBest.R [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  inherit = Archive,
+  public = list(
+
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    #'
+    #' @param check_values (`logical(1)`)\cr
+    #' ignored.
+    initialize = function(search_space, codomain, check_values = FALSE) {
+      self$search_space = assert_param_set(search_space)
```
Comment (jakob-r) [inline R/ArchiveBest.R:25]: wird das nicht in `super` gemacht?

File: R/ArchiveBest.R [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+      if(self$n_evals == 0) {
+        stop("No results stored in archive")
+      } else {
+        cbind(private$.best_x, private$.best_y)
+      }
+    }
+  ),
+
+  active = list(
+
```
Comment (jakob-r) [inline R/ArchiveBest.R:81]: Maybe add a `$data` AB that returns an error if accessed and if we try to overwrite `$data`?

File: R/ArchiveBest.R [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    #' Number of evaluations stored in the archive.
+    n_evals = function() private$.n_evals,
+
+    #' @field n_batch (`integer(1)`)\cr
+    #' Number of batches stored in the archive.
+    n_batch = function() 1,
+
+    #' @field cols_x (`character()`)\cr
+    #' Column names of search space parameters.
+    cols_x = function() self$search_space$ids(),
```
Comment (jakob-r) [inline R/ArchiveBest.R:92]: Simply inherit from archive? i.e.: do not have this lines here?

File: R/ArchiveBest.R [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+    #' @field cols_y (`character()`)\cr
+    #' Column names of codomain parameters.
+    cols_y = function() self$codomain$ids()
```
Comment (jakob-r) [inline R/ArchiveBest.R:96]: see above

**Review by jakob-r — CHANGES_REQUESTED** [review]

File: R/ArchiveBest.R (hunk 3 above)
Comment (be-marc) [reply R/ArchiveBest.R:81]: This does not work if `$data` is already a regular field?

Comment (jakob-r) [reply R/ArchiveBest.R:79]: Could be :) If you know it, I believe you. Otherwise, you have to try it out.

File: R/ArchiveBest.R [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+  active = list(
+
+    #' @field n_evals (`integer(1)`)\cr
+    #' Number of evaluations stored in the archive.
+    n_evals = function() private$.n_evals,
+
+    #' @field n_batch (`integer(1)`)\cr
+    #' Number of batches stored in the archive.
+    n_batch = function() 1
```
Comment (jakob-r) [inline R/ArchiveBest.R:86]: shouldn't `n_batch` also be decreased by one with every call of `add_evals`?

**Comment by be-marc** [comment]
Moved to #121

**Pull request closed without merging**